lib/index.ts

```diff
@@ -0,0 +1,167 @@
+import * as socketio from 'socket.io';
+import * as rword from 'rword';
+import * as Constants from './Constants';
+import * as Types from './Types';
+import * as uuid from 'uuid';
+
+/**
+ * Important Notes:
+ *
+ * UserID is the Socket ID on connection.
+ * This socket id changes on page refreshes and reconnects from React service.
+ */
+
+// States
+
+const displayNames: Types.UserDisplayMap = {};
+const userRooms: Types.UserRoomMap = {};
+const rooms: Types.RoomMap = {};
+
+// Helper functions
+
+/**
+ * Logs message with non-error coloring. Also includes timestamp.
+ *
+ * @param info - message to log
+ */
+function logInfo(info: string): void {
+    console.log('\x1b[33m%s\x1b[0m', `[${new Date().toLocaleString()}]: ${info}`);
+}
+
+/**
+ * Logs message with error coloring. Also includes timestamp.
+ *
+ * @param info - error to log
+ */
+function logError(error: string): void {
+    console.log('\x1b[31m%s\x1b[0m', `[${new Date().toLocaleString()}]: ${error}`);
+}
+
+/**
+ * Returns random hex color
+ */
+function generateRandomColor(): string {
+    return (
+        '#' +
+        Math.random()
+            .toString(16)
+            .slice(2, 8)
+    );
+}
+
+// Socket Connection
+
+const io = socketio();
+io.on('connection', client => {
+    /**
+     * Received when client logs in as guest and sends display name and color
+     */
+    client.on(Constants.GET_DISPLAY_NAME, () => {
```

Comment:
Hmm, it might be nice to make these messages a npm library or some shared file between the 2 apps but that can be for later.

Reply:
Yep, the two files that should be shared are `Types.ts` and `Constants.ts` (socket messaging part). I have another project that does that and I think the easy solution would be to put this repo and `jump-react-service` in the same directory and serve everything through Express (like you initially suggested :') ).

What do you think?

Reply:
That works for me.